src/tiledb/cloud/sql/tiledb_cursor.py

```diff
@@ -0,0 +1,117 @@
+import pyarrow as pa
+
+from tiledb.cloud.sql._execution import exec
+from tiledb.cloud.sql.db_api_exceptions import DataError
+from tiledb.cloud.sql.db_api_exceptions import NotSupportedError
+from tiledb.cloud.sql.db_api_exceptions import ProgrammingError
+
+
+def _getDBType(dtype):
+    if pa.types.is_timestamp(dtype):
+        return "DATETIME"
+    elif pa.types.is_boolean(dtype):
+        return "BOOLEAN"
+    elif pa.types.is_string(dtype):
+        return "STRING"
+    elif pa.types.is_integer(dtype) or pa.types.is_floating(dtype):
+        return "NUMBER"
+    return "UNKNOWN"
+
+
+class Cursor:
+    def __init__(self):
+        self._results = None
+        self._row_index = 0
+        self.arraysize = 1
+        self._description_cache = None
+
+    def executemany(self, query):
+        self.execute(query)
+
+    def execute(self, query):
+        try:
+            self._results = exec(query=query, raw_results=True)
+            self._description_cache = None
+        except Exception as e:
+            self._results = None
+            raise DataError(f"Error executing query: {e}") from e
+
+    def fetchmany(self, size=-1):
+        if self._results is None or len(self._results) == 0:
```

Comment:
These can be simplified to `if not self._results`; sequences of length 0 are considered falsey (here specifically, and also mostly in general).